test/common/upstream/health_checker_impl_test.cc

```diff
@@ -614,6 +635,36 @@ TEST_F(HttpHealthCheckerImplTest, RemoteCloseBetweenChecks) {
   EXPECT_TRUE(cluster_->hosts_[0]->healthy());
 }
 
+// Test that we close connections on a healthy check when reuse_connection is false
```

Comment:
nit: please end all comments with full stops ('.'). Same applies to other stuff in this file obviously.